beater/cloudbeat.go

```diff
@@ -37,6 +37,10 @@ import (
 	"github.com/gofrs/uuid"
 )
 
+const (
+	reconfigureWaitTimeout = 5 * time.Minute
```

Comment:
Instead of using a const, I would get it from the config.
Then, you can add a unit test to assert on the expected behavior when it timed out.